lottie/src/main/java/com/airbnb/lottie/utils/OffscreenLayer.java

```diff
@@ -0,0 +1,494 @@
+      BITMAP,
+      /** Render into a RenderNode's display-list and then draw the render node. (Hardware accelerated) */
+      RENDER_NODE
+  };
+
+  /** Parent render surface should compose onto. null if no rendering is in progress. */
+  @Nullable private Canvas parentCanvas;
+  /** Configuration for compositing the layer onto parentCanvas */
+  @Nullable private ComposeOp op;
+  /** Strategy that we've chosen for rendering this pass */
+  private RenderStrategy currentStrategy;
+  /** Rectangle that the final composition will occupy in the screen */
+  @Nullable private Rect targetRect;
+
+  // For RenderStrategy.SAVE_LAYER:
+  /** Paint passed to Utils.saveLayerCompat(). */
+  @Nullable private Paint composePaint;
+
+  // For RenderStrategy.BITMAP:
+  @Nullable private Bitmap bitmap;
+  @Nullable private Canvas bitmapCanvas;
+  private LPaint clearPaint;
+
+  /** parentCanvas' pre-existing matrix when start() was called */
+  @Nullable float[] preExistingTransform;
+
+  // Android doesn't render shadows on arbitrary bitmaps correctly. Instead, if we're using
+  // render-to-bitmap, we have to draw them manually. To do so, we need an additional bitmap, an
+  // associated canvas, paint, and some other data, which we define as members to avoid
+  // reallocation. (And also because in the render-node case, we won't need them.)
+  @Nullable private Bitmap shadowBitmap;
+  @Nullable private Bitmap shadowMaskBitmap;
+  @Nullable private Canvas shadowBitmapCanvas;
+  @Nullable private Canvas shadowMaskBitmapCanvas;
+  @Nullable private LPaint shadowPaint;
+  @Nullable private BlurMaskFilter shadowBlurFilter;
+  private float lastShadowBlurRadius = 0.0f;
+
+  // For RenderStrategy.RENDER_NODE:
+  @Nullable private RenderNode renderNode; // Render node with the initial contents of the layer
+  @Nullable private RenderNode shadowRenderNode; // Render node for the shadow
+  @Nullable private Canvas renderNodeCanvas;
+  @Nullable private DropShadow lastRenderNodeShadow;
+
+  public OffscreenLayer() {
+    this.clearPaint = new LPaint();
+    this.clearPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.CLEAR));
+  }
+
+  private RenderStrategy chooseRenderStrategy(Canvas parentCanvas, RectF bounds, ComposeOp op) {
+    if (op.isNoop()) {
+      // Can draw directly onto the final canvas, results will look the same.
+      return RenderStrategy.DIRECT;
+    }
+
+    if (!op.hasShadow()) {
+      // Canvas.saveLayer() supports alpha-compositing, blend modes, and color filters, which is
+      // sufficient for this case, and is faster than manually maintaining an off-screen bitmap.
+      // It is not clear if it's faster than RENDER_NODE and when, but this is what we've been
+      // doing prior to OffscreenLayer, so keep that behavior to be safe and avoid regressions.
+      return RenderStrategy.SAVE_LAYER;
+    }
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q && parentCanvas.isHardwareAccelerated()) {
```

Comment:
nit: I like how you have the two guard cases above. I think having the BITMAP case at the end instead be another guard statement here, it would be easier to read. It'll also allow you un-nest this nested if.